### DISCUSSION TITLE
Question/Request: Define Multiple PVC to support different tablespaces on different storageClasses

### DISCUSSION THREAD

**Discussion by adminnz** [cloudnative-pg/cloudnative-pg#1152] [Ideas]
I'm looking at using CloudNative-PG to run a number of timescaledb clusters (still working on getting the docker image running).
But one thing that has come up from our team is if its possible to define a cluster with multiple PVC mounted to different paths that can then (ideally with the cluster CRD, but also fine if its manually) be used for different tablespaces.

This would allow us to define for example, a tablespace with a mount to /data/0 using a performant storageClass (we are in Azure, so maybe premium-ssd) but then migrate data to a different tablespace with a mount to /data/1 using a more cost effective storageClass (e.g. standard-hdd).

Is this currently possible? and if not, could it be added?

**Comment by gbartolini** [comment]
Hi @adminnz,

your request is legitimate. At the moment, this is the biggest limitation in terms of vertical scalability of CloudNativePG. While you can create table partitions for horizontal partitioning, you cannot scale on the I/O side unless you stripe the underlying disks. Tablespaces are also important for indexes. So, I hear you.

However, in our view, when you manage tablespaces in a cloud native space, you also need to worry not just about setting them up, but also about backups, recovery, self-healing, and so on. This also involves temporary tablespaces (to scale sort operations for example and joins).

It's no surprise that we have an epic for this task in our future plans: https://github.com/cloudnative-pg/cloudnative-pg/issues/11

Please look at the proposed interface. We do not have an immediate plan to work on that feature with our current capacity, but we always welcome contributions.

**Comment by adminnz** [comment]
I might not have explained the use-case well enough. 
While disk-striping for IO performance is definately a use-case. 
Our use-case is more for using timescaledb's features to move data from one tablespace to another for the purpose of keeping hot data on performant storage and cool data on more cost effective storage.
We do not have a huge depth of knowledge in postgres internals but from what we understand pgbarman supports tablespace backup & recovery.

**Reply by gbartolini** [reply]
Hi @adminnz 

I perfectly understand the use case and I confirm that it is not just a matter of mounting volumes unfortunately.

Tablespaces have some basic management tasks to be done: backup and recovery must be taken into account as part of that feature development, in particular as far testing is concerned - even though as you say barman-cloud-backup and barman-cloud-restore already support tablespaces and tablespace remapping.
